pkg/operator/serviceaccountissuercontroller/serviceaccountissuer_controller.go

```diff
@@ -60,31 +73,27 @@ func (c *ServiceAccountIssuerController) sync(ctx context.Context, controllerCon
 
 	// there is no service account issuer set and there are no service account issuers in status, no-op.
 	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
-		return nil
-	}
-
-	// if the status is not set, but the current issuer is not empty, update the status and let the controller resync
-	if len(currentIssuer) > 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
 		operatorCopy := operator.DeepCopy()
-		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{
-			{
-				Name: currentIssuer,
-			},
-		}
+		operatorCopy.Status.ServiceAccountIssuers = defaultServiceAccountIssuerValue
 		_, statusUpdateErr := c.kubeAPIServerOperatorClient.UpdateStatus(ctx, operatorCopy, metav1.UpdateOptions{})
 		if statusUpdateErr == nil {
-			klog.Infof("ServiceAccountIssuer %q is now set as active issuer", currentIssuer)
+			klog.Infof("ServiceAccountIssuer %q is set as a default value", defaultServiceAccountIssuerValue[0].Name)
 			statusUpdateErr = factory.SyntheticRequeueError
 		}
 		return statusUpdateErr
 	}
 
-	// in this case, user deleted service account issuer value, which means we have to remove all values from the status
-	// if we don't do this and instead just remove first value, the new tokens will be issued with "previous" service
-	// account issuer which is not desired
-	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) != 0 {
+	// this is a case when issuer is not set in auth config and the operator status already has the default issuer set.
+	if isDefaultServiceAccountIssuer(currentIssuer, operator.Status.ServiceAccountIssuers) {
+		return nil
+	}
+
+	// in this case the issuer was previously set in auth config, and later it was removed.
+	// this case must result into having default issuer set.
+	// this allows to instantly remove compromised issuers if needed.
+	if len(currentIssuer) == 0 {
 		operatorCopy := operator.DeepCopy()
-		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{}
+		operatorCopy.Status.ServiceAccountIssuers = defaultServiceAccountIssuerValue
```

Comment:
We cannot assume this, no? The previous issuer must still be trusted or your cluster gets bricked when you decide that you don't need to have your own issuer.

Reply:
@stlaz i believe this is the corner case of "my cluster was hacked and I need to change the issuer right now". in that case, people might not care about bricking their cluster for ~1h if they can invalidate all tokens immediately

Reply:
We do need some kind of a revocation mechanism, true. This should be documented in the API, though.

Reply:
@stlaz i plan to update the product documentation with this (as was agreed in API review with Joel)

Reply:
This must be a part of the API documentation as well. It's an action taken per a specific API field value, if it's not documented in the API, it is a bug.

Reply:
@stlaz separate, existing bug :-) feel free to jira it.